front/lib/resources/user_resource.ts

```diff
@@ -0,0 +1,147 @@
+import type { Result, UserProviderType, UserType } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import type { Attributes, ModelStatic, Transaction } from "sequelize";
+
+import { renderUserType } from "@app/lib/api/user";
+import type { Authenticator } from "@app/lib/auth";
+import { User } from "@app/lib/models/user";
+import { BaseResource } from "@app/lib/resources/base_resource";
+import { MembershipModel } from "@app/lib/resources/storage/models/membership";
+import type { ReadonlyAttributesType } from "@app/lib/resources/storage/types";
+
+// Attributes are marked as read-only to reflect the stateless nature of our Resource.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface, @typescript-eslint/no-unsafe-declaration-merging
+export interface UserResource extends ReadonlyAttributesType<User> {}
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
+export class UserResource extends BaseResource<User> {
+  static model: ModelStatic<User> = User;
+
+  constructor(model: ModelStatic<User>, blob: Attributes<User>) {
+    super(User, blob);
+  }
+
+  static async findAllByUsername(username: string): Promise<UserResource[]> {
+    const users = await User.findAll({
+      where: {
+        username,
+      },
+    });
+
+    return users.map((user) => new UserResource(User, user.get()));
+  }
+
+  static async fetchByExternalId(
+    sid: string,
+    transaction?: Transaction
```

Comment:
We generally don't want transactions in the interface of Resources cc @flvndvd

Reply:
I presume there's no way around all these for now... They will be removable as we have higher and higher level resources

Reply:
Actually this one can be removed from reviewing the rest of the codebase